packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -4642,4 +4667,53 @@ export class SnapController extends BaseController<
       runtime.state = undefined;
     }
   }
+
+  /**
+   * Retrieve the source code for a snap from storage.
+   *
+   * @param snapId - The snap ID.
+   * @returns The source code for the snap.
+   */
+  async #getSourceCode(snapId: SnapId) {
+    const { result } = await this.messenger.call(
```

Comment:
Can the usage of the StorageService throw?

Reply:
not for `getItem`